eox_nelp/stats/tests/api/v1/tests_views.py

```diff
@@ -0,0 +1,230 @@
+"""
+from ddt import data, ddt
+from django.contrib.sites.models import Site
+from django.test import override_settings
+from django.urls import reverse
+from mock import Mock, patch
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+
+@ddt
+class GeneralTenantStatsViewTestCase(APITestCase):
+    """ Test GeneralTenantStatsView."""
+
+    def setUp(self):
+        """
+        Create site since the view use the request.site attribute to determine the current domain.
+        """
+        Site.objects.get_or_create(domain="testserver")
+
+    @override_settings(MIDDLEWARE=["eox_tenant.middleware.CurrentSiteMiddleware"])
+    @patch("eox_nelp.stats.api.v1.views.metrics")
+    def test_default(self, mock_metrics):
+        """
+        Test a get request, this will verify the standard view behavior by checking the call of the metrics functions.
+
+        Expected behavior:
+            - Response data contains the expected keys.
+            - Status code 200.
+        """
+        mock_metrics.get_learners_metric.return_value = 5
+        mock_metrics.get_instructors_metric.return_value = 4875
+        mock_metrics.get_courses_metrics.return_value = {
+            "total_courses": 3,
+            "metrics": [
+                {}, {}, {}
+            ],
+        }
+        url_endpoint = reverse("stats-api:v1:general-stats")
+
+        response = self.client.get(url_endpoint)
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertTrue(["learners", "courses", "instructors", "components"] == list(response.data.keys()))
+
+    @override_settings(
+        MIDDLEWARE=["eox_tenant.middleware.CurrentSiteMiddleware"],
+        STATS_SETTINGS={"API_XBLOCK_TYPES": ["html", "problem", "video"]},
+    )
+    @patch("eox_nelp.stats.api.v1.views.metrics")
+    def test_total_components(self, mock_metrics):
+        """
+        Test that the view will calculate the total of components based on the metrics values
+
+        Expected behavior:
+            - Status code 200.
+            - Components total values are the expected.
+        """
+        fake_metric = {
+            "components": {
+                "html": 5,
+                "problem": 10,
+                "video": 15,
+            }
+        }
+        mock_metrics.get_learners_metric.return_value = 5
+        mock_metrics.get_instructors_metric.return_value = 4875
+        mock_metrics.get_courses_metrics.return_value = {
+            "total_courses": 3,
+            "metrics": [
+                fake_metric, fake_metric, fake_metric, fake_metric
+            ],
+        }
+        url_endpoint = reverse("stats-api:v1:general-stats")
+
+        response = self.client.get(url_endpoint)
+
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
+        self.assertEqual({"html": 20, "problem": 40, "video": 60}, response.data["components"])
```

Comment:
This could be something like
```suggestion
        self.assertEqual({"html": len(mock_metrics.get_courses_metrics())*fake_metric["html"], "problem": 40, "video": 60}, response.data["components"])
```
Also may be could be a dict comprehension
``` python
{key:value*len(mock_metrics.get_courses_metrics()) for (key, value) in fake_metric["components"].items()}
```

Reply:
when you executes a mock like `mock_metrics.get_courses_metrics()` you are increasing the number of calls  that affects some tests